src/palace/manager/api/overdrive/coverage.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from sqlalchemy.orm import Session
+
+from palace.manager.api.overdrive.constants import OVERDRIVE_LABEL
+from palace.manager.api.overdrive.representation import OverdriveRepresentationExtractor
+from palace.manager.core.coverage import BibliographicCoverageProvider, CoverageFailure
+from palace.manager.core.metadata_layer import Metadata
+from palace.manager.sqlalchemy.model.collection import Collection
+from palace.manager.sqlalchemy.model.datasource import DataSource
+from palace.manager.sqlalchemy.model.identifier import Identifier
+
+if TYPE_CHECKING:
+    from palace.manager.api.overdrive.api import OverdriveAPI
+
+
+class OverdriveBibliographicCoverageProvider(BibliographicCoverageProvider):
+    """Fill in bibliographic metadata for Overdrive records.
+
+    This will occasionally fill in some availability information for a
+    single Collection, but we rely on Monitors to keep availability
+    information up to date for all Collections.
+    """
+
+    SERVICE_NAME = "Overdrive Bibliographic Coverage Provider"
+    DATA_SOURCE_NAME = DataSource.OVERDRIVE
+    PROTOCOL = OVERDRIVE_LABEL
+    INPUT_IDENTIFIER_TYPES = Identifier.OVERDRIVE_ID
+
+    def __init__(
+        self,
+        collection: Collection,
+        api: OverdriveAPI | None = None,
```

Comment:
This used to be:
```python
api_class: type[OverdriveAPI] | OverdriveAPI = OverdriveAPI,
```

I updated (and renamed the parameter) to break an import cycle with `OverdriveAPI`. This is probably the biggest functional change I made in this PR, but I didn't see a nicer way to make the imports work. 

It looks like this parameter is only used in tests and one place in `OverdriveAPI`, so I think its safe to update.